File: 프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people, limit):
    answer = 0
    
    
    boat = []
    people.sort(reverse=True)
    i = 0
    j = len(people)-1
    while i<=j:
        if people[i] + people[j] <= limit:
            i += 1
            j -= 1
        else:
            i+= 1
        answer += 1
    
    
    
    
    return answer

File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        
        Arrays.sort(people);
        
        int left = 0;
        int right = people.length-1;
        
        while (left <= right) {
            if (people[right] + people[left] <= limit) {
                right -= 1;
                left += 1;
            } else {
                right -= 1;
            }
            answer += 1;
        }
        return answer;
    }
}